clippy_lints/src/zero_sized_map_values.rs

```diff
@@ -0,0 +1,152 @@
+use if_chain::if_chain;
+use rustc_hir::{self as hir, intravisit::FnKind, *};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty::{Adt, Ty};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::Span;
+use rustc_target::abi::LayoutOf as _;
+use rustc_typeck::hir_ty_to_ty;
+
+use crate::utils::{match_type, paths, span_lint_and_help};
+
+declare_clippy_lint! {
+    /// **What it does:** Checks for maps with zero-sized value types anywhere in the code.
+    ///
+    /// **Why is this bad?** Since there is only a single value for a zero-sized type, a map
+    /// containing zero sized values is effectively a set. Using a set in that case improves
+    /// readability and communicates intent more clearly.
+    ///
+    /// **Known problems:** A zero-sized type cannot be recovered later if it contains private
```

Comment:
As highlighted in [this comment](https://github.com/rust-lang/rust-clippy/pull/6165#pullrequestreview-510993250), we should not lint in public item signatures, or leave that for a follow-up and mention that in the `Known problems` section.

If we go for the latter, then the category should be `pedantic` instead of `complexity`.

Reply:
* Updated known problems
* Changed category to pedantic